src/test/scala/edu/berkeley/ce/rockslicing/RockSlicerSpec.scala

```diff
@@ -205,157 +204,212 @@ class RockSlicerSpec extends FunSuite {
+   val expectedBlocksCentroid = expectedBlocks map { block =>
+     val centroid = block.centroid
+     Block(centroid, block.updateFaces(centroid))
+   }
+
+   val blockCheck = centroidMergedBlocks.zip(expectedBlocksCentroid) map { case (calc, expected) =>
+     calc.approximateEquals(expected)
+   }
+   assert(orphanBlocks.isEmpty)
+   assert(centroidMergedBlocks.length == 2)
+   assert(!blockCheck.contains(false))
+ }
+
+ test("Processor joints should be removed and actual blocks restored") {
+   val processorJoint1 = Joint((0.0, 1.0, 0.0), (0.0, 0.0, 0.0),
+                               (0.0, 0.3, 0.0), phi = 0.0, cohesion = 0.0, shape = Nil,
+                                processorJoint = true)
+   val processorJoint2 = Joint((0.0, 1.0, 0.0), (0.0, 0.0, 0.0),
+                               (0.0, 0.7, 0.0), phi = 0.0, cohesion = 0.0, shape = Nil,
+                                processorJoint = true)
+   val processorJoint3 = Joint((0.0, 1.0, 0.0), (0.0, 0.0, 0.0),
+                               (0.0, 0.2, 0.0), phi = 0.0, cohesion = 0.0, shape = Nil,
+                               processorJoint = true)
+   val processorJoint4 = Joint((0.0, 1.0, 0.0), (0.0, 0.0, 0.0),
+                               (0.0, 0.8, 0.0), phi = 0.0, cohesion = 0.0, shape = Nil,
+                               processorJoint = true)
+
+   val actualJoint1 = Joint((0.0, 1.0, 0.0), (0.0, 0.0, 0.0), (0.0, 0.25, 0.0), phi = 0.0,
+                            cohesion = 0.0, shape = Nil)
+   val actualJoint2 = Joint((0.0, 1.0, 0.0), (0.0, 0.0, 0.0), (0.0, 0.75, 0.0), phi = 0.0,
+                            cohesion = 0.0, shape = Nil)
+   val joints = Seq(processorJoint1, processorJoint2, processorJoint3, processorJoint4,
+                    actualJoint1, actualJoint2)
+   var blocks = Seq(unitCube)
+   for (joint <- joints) {
+     blocks = blocks.flatMap(_.cut(joint))
+   }
+
+   val nonRedundantBlocks = blocks.map { case block @ Block(center, _) =>
+     Block(center, block.nonRedundantFaces)
+   }
+
+   val processorBlocks = nonRedundantBlocks.filter { block =>
+     block.faces.exists { face => face.processorJoint }
+   }
+   val (mergedBlocks, orphanBlocks) = RockSlicer.mergeBlocks(processorBlocks, Seq.empty[Block],
+                                                             Seq.empty[Block])
+
+   val centroidMergedBlocks = mergedBlocks.map{ block =>
+     val centroid = block.centroid
+     Block(centroid, block.updateFaces(centroid))
+   }
+
+   val expectedBlocks = Seq(rightQuarterUnitCube, centerPartUnitCube, leftQuarterUnitCube)
+   val expectedBlocksCentroid = expectedBlocks map { block =>
+     val centroid = block.centroid
+     Block(centroid, block.updateFaces(centroid))
+   }
+
+   val blockCheck = centroidMergedBlocks.zip(expectedBlocksCentroid) forall { case (calc, expected) =>
+     calc.approximateEquals(expected)
+   }
+
+   assert(orphanBlocks.isEmpty)
+   assert(centroidMergedBlocks.length == 3)
+   assert(blockCheck)
```

Comment:
Now here, you can just avoid saving a `val` and do the assert on the line where you originally created it. Just use the right-hand side of the assignment as the argument to `assert`.